images/minimal-notebook/setup-scripts/setup_julia.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+# Requirements:
+# - Run as the root user
+# - The JULIA_PKGDIR environment variable is set
+
+import os
+import platform
+import shutil
+import subprocess
+from pathlib import Path
+
+import requests
+
+
+def unify_aarch64(platform: str) -> str:
+    return {
+        "aarch64": "aarch64",
+        "arm64": "aarch64",  # To support local building on aarch64 Macs
+        "x86_64": "x86_64",
+    }[platform]
+
+
+def get_latest_julia_url() -> tuple[str, str]:
+    """
+    Get the last stable version of Julia
+    Based on: https://github.com/JuliaLang/www.julialang.org/issues/878#issuecomment-749234813
+    """
+
+    versions = requests.get(
+        "https://julialang-s3.julialang.org/bin/versions.json"
+    ).json()
+    stable_versions = {k: v for k, v in versions.items() if v["stable"]}
+    latest_version_files = stable_versions[max(stable_versions)]["files"]
+    triplet = unify_aarch64(platform.machine()) + "-linux-gnu"
+    file_info = [vf for vf in latest_version_files if vf["triplet"] == triplet][0]
+    return file_info["url"], file_info["version"]
+
+
+def download_julia(julia_url: str) -> None:
+    """
+    Downloads and unpacks julia
+    The resulting julia directory is "/opt/julia-VERSION/"
+    """
+    tmp_file = Path("/tmp/julia.tar.gz")
+    subprocess.check_call(
+        ["curl", "--progress-bar", "--location", "--output", tmp_file, julia_url]
+    )
+    shutil.unpack_archive(tmp_file, "/opt/")
+    tmp_file.unlink()
+
+
+def prepare_julia(julia_version: str) -> None:
+    """
+    Creates /usr/local/bin/julia symlink
+    Make Julia aware of conda libraries
+    Creates a directory for Julia user libraries
+    """
+    # Link Julia installed version to /usr/local/bin, so julia launches it
+    subprocess.check_call(
+        ["ln", "-fs", f"/opt/julia-{julia_version}/bin/julia", "/usr/local/bin/julia"]
+    )
+
+    # Tell Julia where conda libraries are
+    Path("/etc/julia").mkdir()
```

Comment:
This will fail if this directory already exists, which I think is ok - we don't need to pass `exists_ok=True` here.

Reply:
It's better to fail and know about it, because I don't expect this dir to exist.